streams.go

```diff
@@ -366,3 +363,46 @@ func (r *ByteStreamReader) ReadAll() []byte {
 type TextStreamHandler func(reader *TextStreamReader, participantIdentity string)
 
 type ByteStreamHandler func(reader *ByteStreamReader, participantIdentity string)
+
+// ---------------------------------------------------------
+
+func chunkUtf8String(s string) [][]byte {
+	chunks := [][]byte{}
+	stringBytes := []byte(s)
+
+	for len(stringBytes) > STREAM_CHUNK_SIZE {
+		k := STREAM_CHUNK_SIZE
```

Comment:
should this be `k := STREAM_CHUNK_SIZE - 1` , i. e. indexing into last element of `STREAM_CHUNK_SIZE` array?

Reply:
If I understand correctly, the idea is that we do not want to split at a continuation byte. So if `k` starts from  `STREAM_CHUNK_SIZE`, and we determine if it is safe to split there (i.e. the string before it is a valid utf-8 string).

When we break out of the loop at `k == STREAM_CHUNK_SIZE` i.e. in the first iteration itself, we get to use the full chunk size available to us. The chunk is formed as `stringBytes[:k]` meaning the `k`th element is not included. 

If we start `k` from `STEAM_CHUNK_SIZE - 1`, that would theoretically work as well, but it's not maxing out on the available max chunk size.

Reply:
got it, makes sense